### PR TITLE
CLN: remove never-used branch from Series ops

### PR DESCRIPTION
**Pull request by jbrockmendel**
_Almost_ gets to a point where Series and DataFrame ops can share the na_op definition.

AFAICT this is the only non-test usage of libalgos.arrmap_object.

### CODE REVIEW

File: pandas/core/ops.py [hunk 1]
```diff
@@ -1713,10 +1713,6 @@ def na_op(x, y):
             result = expressions.evaluate(op, str_rep, x, y, **eval_kwargs)
         except TypeError:
             result = masked_arith_op(x, y, op)
-        except Exception:  # TODO: more specific?
```
Comment (jreback) [inline pandas/core/ops.py]: this seems to be the only usage, let's remove arrmap_object as well

**Review by jreback — CHANGES_REQUESTED** [review]

File: pandas/core/ops.py (hunk 1 above)
Comment (jreback) [reply pandas/core/ops.py]: ahh I c, this is parametrized, ok then.

**Review by jreback — APPROVED** [review]

File: pandas/core/ops.py (hunk 1 above)
Comment (jreback) [reply pandas/core/ops.py]: actually do we use ``arrmap_*`` *anywhere*?

**Review by jreback — CHANGES_REQUESTED** [review]
-revising again; looks like we can completely remove arrmap_*-

nope looks like 1 usage (after you remove this one)

```
(pandas) bash-3.2$ grep -r arrmap_ pandas --include '*.py'
pandas/core/algorithms.py:        return algos.arrmap_float64(q, _get_score)
pandas/core/ops.py:                return libalgos.arrmap_object(x, lambda val: op(val, y))
pandas/tests/test_algos.py:    result = libalgos.arrmap_object(values, lambda x: x in ["foo", "bar"])
(pandas) bash-3.2$ grep -r arrmap_ pandas --include '*.pyx'
pandas/_libs/algos.pyx:arrmap_float64 = arrmap["float64_t"]
pandas/_libs/algos.pyx:arrmap_float32 = arrmap["float32_t"]
pandas/_libs/algos.pyx:arrmap_object = arrmap["object"]
pandas/_libs/algos.pyx:arrmap_int64 = arrmap["int64_t"]
pandas/_libs/algos.pyx:arrmap_int32 = arrmap["int32_t"]
pandas/_libs/algos.pyx:arrmap_uint64 = arrmap["uint64_t"]
pandas/_libs/algos.pyx:arrmap_bool = arrmap["uint8_t"]
```

**Comment by jbrockmendel** [comment]
So what's the conclusion re arrmap_*?

**Comment by jreback** [comment]
```
        q = np.asarray(q, np.float64)
        return algos.arrmap_float64(q, _get_score)
```

I think you can just do ``np.vectorize`` on this and it will work, if so, then let's completely blow away arrmap

**Comment by jbrockmendel** [comment]
I'll open an issue to look into this more carefully.

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
cool, ping on green.

**Comment by jbrockmendel** [comment]
ping

**Comment by jreback** [comment]
thanks @jbrockmendel

**Pull request merged**